src/libstd/sync/mpsc/spsc_queue.rs

```diff
@@ -99,7 +99,9 @@ impl<T, ProducerAddition, ConsumerAddition> Queue<T, ProducerAddition, ConsumerA
     ) -> Self {
         let n1 = Node::new();
         let n2 = Node::new();
-        (*n1).next.store(n2, Ordering::Relaxed);
+        // SAFETY: At this point we know the pointer is not NUL since it was
+        // build just above.
```

Comment:
The pointer must be valid for asserting as a shared reference (which is what the unsafe code does) -- that's true in this case, but we shouldn't really be talking about nullity as the condition which must be satisfied.

Reply:
Fixed ! (I hope)